tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2752,129 +2923,146 @@ private[optimizer] abstract class OptimizerCore(
-              BinaryOp(innerOp @ (Float_+ | Float_-), FloatLiteral(y), z)) =>
-            foldBinaryOp(innerOp, FloatLiteral(x+y), z)
+          case (PreTransLit(FloatLiteral(x)),
+              PreTransBinaryOp(innerOp @ (Float_+ | Float_-),
+                  PreTransLit(FloatLiteral(y)), z)) =>
+            foldBinaryOp(innerOp, PreTransLit(FloatLiteral(x + y)), z)
 
           case _ => default
         }
 
       case Float_- =>
         (lhs, rhs) match {
-          case (_, FloatLiteral(r)) => foldBinaryOp(Float_+, lhs, FloatLiteral(-r))
-
-          case (FloatLiteral(x), BinaryOp(Float_+, FloatLiteral(y), z)) =>
-            foldBinaryOp(Float_-, FloatLiteral(x-y), z)
-          case (FloatLiteral(x), BinaryOp(Float_-, FloatLiteral(y), z)) =>
-            foldBinaryOp(Float_+, FloatLiteral(x-y), z)
-
-          case (_, BinaryOp(BinaryOp.Float_-, FloatLiteral(0), x)) =>
+          case (_, PreTransLit(FloatLiteral(r))) =>
+            foldBinaryOp(Float_+, lhs, PreTransLit(FloatLiteral(-r)))
+
+          case (PreTransLit(FloatLiteral(x)),
+              PreTransBinaryOp(Float_+, PreTransLit(FloatLiteral(y)), z)) =>
+            foldBinaryOp(Float_-, PreTransLit(FloatLiteral(x - y)), z)
+          case (PreTransLit(FloatLiteral(x)),
+              PreTransBinaryOp(Float_-, PreTransLit(FloatLiteral(y)), z)) =>
+            foldBinaryOp(Float_+, PreTransLit(FloatLiteral(x - y)), z)
+
+          case (_, PreTransBinaryOp(BinaryOp.Float_-,
+              PreTransLit(FloatLiteral(0)), x)) =>
             foldBinaryOp(Float_+, lhs, x)
 
           case _ => default
         }
 
       case Float_* =>
         (lhs, rhs) match {
-          case (FloatLiteral(l), FloatLiteral(r)) => FloatLiteral(l * r)
-          case (_, FloatLiteral(_))               => foldBinaryOp(Float_*, rhs, lhs)
+          case (_, PreTransLit(FloatLiteral(_))) =>
+            foldBinaryOp(Float_*, rhs, lhs)
 
-          case (FloatLiteral(1), _)  => rhs
-          case (FloatLiteral(-1), _) => foldBinaryOp(Float_-, FloatLiteral(0), rhs)
+          case (PreTransLit(FloatLiteral(1)), _) =>
+            rhs
+          case (PreTransLit(FloatLiteral(-1)), _) =>
+            foldBinaryOp(Float_-, PreTransLit(FloatLiteral(0)), rhs)
 
           case _ => default
         }
 
       case Float_/ =>
         (lhs, rhs) match {
-          case (FloatLiteral(l), FloatLiteral(r)) => FloatLiteral(l / r)
-
-          case (_, FloatLiteral(1))  => lhs
-          case (_, FloatLiteral(-1)) => foldBinaryOp(Float_-, FloatLiteral(0), lhs)
+          case (_, PreTransLit(FloatLiteral(1))) =>
+            lhs
+          case (_, PreTransLit(FloatLiteral(-1))) =>
+            foldBinaryOp(Float_-, PreTransLit(FloatLiteral(0)), lhs)
 
           case _ => default
         }
 
       case Float_% =>
         (lhs, rhs) match {
-          case (FloatLiteral(l), FloatLiteral(r)) => FloatLiteral(l % r)
-          case _                                  => default
+          case _ => default
```

Comment:
Oh it won't. At least not before the new optimizer of 2.12. But I don't care. It's a rare case anyway. I just want my cases to be consistent.